### PR TITLE
Feature lwdev 4794 add models adapter

### PR DESCRIPTION
**Pull request by nikagamkrelidze**

### CODE REVIEW

File: client/Lykke.Service.OperationsHistory.Client/Lykke.Service.OperationsHistory.Client.csproj [hunk 1]
```diff
@@ -16,16 +15,12 @@
...
-
   <ItemGroup>
     <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.2.0" />
-    <PackageReference Include="AutoMapper" Version="6.2.1" />
-    <PackageReference Include="Lykke.Common" Version="3.0.2" />
+    <PackageReference Include="Lykke.Common" Version="4.3.0" />
     <PackageReference Include="Microsoft.Rest.ClientRuntime" Version="2.3.10" />
   </ItemGroup>
-
   <ItemGroup>
```
Comment (MikhailGerasimov) [inline client/Lykke.Service.OperationsHistory.Client/Lykke.Service.OperationsHistory.Client.csproj]: Enable XML documentation for Release configuration

File: src/Lykke.Service.OperationsHistory.Core/Settings/Api/ApiSettings.cs [hunk 2]
```diff
@@ -39,4 +40,9 @@ public class AzureQueueSettings
 
         public string QueueName { get; set; }
     }
+
+    public class AssetsServiceClientSettings
+    {
+        public string ServiceUrl { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory.Core/Settings/Api/ApiSettings.cs]: Add a validation attribute

Comment (MikhailGerasimov) [reply src/Lykke.Service.OperationsHistory.Core/Settings/Api/ApiSettings.cs]: For all external services

File: src/Lykke.Service.OperationsHistory/LegacyOperationsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,122 @@
+﻿using Common;
+using Lykke.Service.Assets.Client.Models;
+using Lykke.Service.OperationsHistory.Core;
+using Lykke.Service.OperationsRepository.Contract;
+using Lykke.Service.OperationsRepository.Contract.Abstractions;
+using System;
+
+namespace Lykke.Service.OperationsHistory
+{
+    public static class LegacyOperationsExtensions
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/LegacyOperationsExtensions.cs]: Could be internal

File: src/Lykke.Service.OperationsHistory/Models/HistoryOperationModel.cs [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        public string Id { get; set; }
+        public double Amount { get; set; }
+        public string DateTime { get; set; }
+        public string Asset { get; set; }
+        public string BlockChainHash { get; set; }
+        public bool IsRefund { get; set; }
+        public string AddressFrom { get; set; }
+        public string AddressTo { get; set; }
+        public bool IsSettled { get; set; }
+        public string Type { get; set; }
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Models/HistoryOperationModel.cs]: Why all *Types in the mode have string type? Why not enum?

File: src/Lykke.Service.OperationsHistory/Modules/ServiceModule.cs [hunk 5]
```diff
@@ -93,6 +105,28 @@ private void RegisterAzureRepositories(ContainerBuilder builder)
         private void RegisterServiceClients(ContainerBuilder builder)
         {
             builder.RegisterLykkeServiceClient(_settings.ClientAccountServiceClient.ServiceUrl);
+            builder.RegisterInstance<IAssetsService>(
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Modules/ServiceModule.cs]: AssetService has it owns extension for registration?

File: src/Lykke.Service.OperationsHistory/Modules/ServiceModule.cs [hunk 6]
```diff
@@ -93,6 +105,28 @@ private void RegisterAzureRepositories(ContainerBuilder builder)
...
+        }
+
+        private void RegisterDictionaryEntities(ContainerBuilder builder)
+        {
+            builder.Register(c =>
+            {
+                var ctx = c.Resolve<IComponentContext>();
+                return new CachedDataDictionary<string, Asset>(
+                    async () =>
+                        (await ctx.Resolve<IAssetsService>().AssetGetAllAsync()).ToDictionary(itm => itm.Id));
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Modules/ServiceModule.cs]: Why not IAssetsServiceCached?

File: src/Lykke.Service.OperationsHistory/Services/HistoryFilterPredicates.cs [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        public static Func<HistoryOperation, bool> IfTypeEquals(HistoryOperationType? operationType)
+        {
+            return operation => operationType == null || operation.Type == operationType;
+        }
+
+        public static Func<HistoryOperation, bool> IfAssetEquals(string assetId)
+        {
+            return operation =>
+            {
+                if (string.IsNullOrWhiteSpace(assetId)) return true;
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Services/HistoryFilterPredicates.cs]: Is it correct that empty string==true?

File: src/Lykke.Service.OperationsHistory/Services/HistoryMessageAdapter.cs [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+    public class HistoryMessageAdapter : IHistoryMessageAdapter
+    {
+        private readonly IHistoryOperationAdapter _operationAdapter;
+
+        public HistoryMessageAdapter(IHistoryOperationAdapter operationAdapter)
+        {
+            _operationAdapter = operationAdapter;
+        }
+
+        public async Task<HistoryOperation> Execute(OperationsHistoryMessage src)
```
Comment (MikhailGerasimov) [inline src/Lykke.Service.OperationsHistory/Services/HistoryMessageAdapter.cs]: Add Async suffix. Delete async await

File: src/Lykke.Service.OperationsHistory/LegacyOperationsExtensions.cs [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+
+        private static double Normalize(this double amount, Asset asset, bool toUpper = false)
+        {
+            return amount.TruncateDecimalPlaces(asset?.GetDisplayAccuracy() ?? _assetDefaultDisplayAccuracy, toUpper);
+        }
+        
+        private static HistoryOperationState GetState(TransactionStates state)
+        {
+            if (state == TransactionStates.InProcessOffchain ||
+                state == TransactionStates.InProcessOffchain)
```
Comment (tarurar) [inline src/Lykke.Service.OperationsHistory/LegacyOperationsExtensions.cs]: InProcessOnchain instead?

File: src/Lykke.Service.OperationsHistory/LegacyOperationsExtensions.cs [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+                operation.AssetPairId,
+                operation.Price);
+        }
+
+        public static HistoryOperation ConvertToHistoryOperation(this ILimitTradeEvent operation, Asset asset)
+        {
+            var isBuy = operation.OrderType == OrderType.Buy;
+
+            var volume = operation.Volume.Normalize(asset, isBuy);
+            Console.WriteLine(operation.Status.ToString());
```
Comment (tarurar) [inline src/Lykke.Service.OperationsHistory/LegacyOperationsExtensions.cs]: Please remove it

File: tests/Lykke.Service.OperationsHistory.Tests/InMemoryCacheTests.cs [hunk 11]
```diff
@@ -1,128 +0,0 @@
...
-using AutoMapper;
-using Lykke.Service.OperationsHistory.Core.Entities;
-using Lykke.Service.OperationsHistory.Services.InMemoryCache;
-using Microsoft.VisualStudio.TestTools.UnitTesting;
-using Moq;
-
-namespace Lykke.Service.OperationsHistory.Tests
-{
-    [TestClass]
-    public class InMemoryCacheTests
```
Comment (tarurar) [inline tests/Lykke.Service.OperationsHistory.Tests/InMemoryCacheTests.cs]: Why deleted tests at all?

**Pull request merged**